include/nbl/builtin/hlsl/bxdf/reflection/lambertian.hlsl

```diff
@@ -0,0 +1,176 @@
+    Scalar getNdotL() NBL_CONST_MEMBER_FUNC { return hlsl::mix(math::conditionalAbsOrMax<Scalar>(_clamp == BxDFClampMode::BCM_ABS, _sample.getNdotL(), 0.0), _sample.getNdotL(), _clamp == BxDFClampMode::BCM_NONE); }
+    Scalar getNdotLUnclamped() NBL_CONST_MEMBER_FUNC { return _sample.getNdotL(); }
+    Scalar getNdotL2() NBL_CONST_MEMBER_FUNC { return _sample.getNdotL2(); }
+    Scalar getVdotL() NBL_CONST_MEMBER_FUNC { return _sample.getVdotL(); }
+
+    // aniso
+    Scalar getTdotL2() NBL_CONST_MEMBER_FUNC { return _sample.getTdotL() * _sample.getTdotL(); }
+    Scalar getBdotL2() NBL_CONST_MEMBER_FUNC { return _sample.getBdotL() * _sample.getBdotL(); }
+    Scalar getTdotV2() NBL_CONST_MEMBER_FUNC { return interaction.getTdotV() * interaction.getTdotV(); }
+    Scalar getBdotV2() NBL_CONST_MEMBER_FUNC { return interaction.getBdotV() * interaction.getBdotV(); }
+
+    LS _sample;
+    SI interaction;
+    BxDFClampMode _clamp;
+};
+
+template<class LS, class Iso, class Aniso, class Spectrum NBL_PRIMARY_REQUIRES(LightSample<LS> && surface_interactions::Isotropic<Iso> && surface_interactions::Anisotropic<Aniso>)
+struct SLambertianBxDF
+{
+    using this_t = SLambertianBxDF<LS, Iso, Aniso, Spectrum>;
+    using scalar_type = typename LS::scalar_type;
+    using ray_dir_info_type = typename LS::ray_dir_info_type;
+    using isotropic_interaction_type = Iso;
+    using anisotropic_interaction_type = Aniso;
+    using sample_type = LS;
+    using spectral_type = Spectrum;
+    using quotient_pdf_type = sampling::quotient_and_pdf<spectral_type, scalar_type>;
+
+    using params_isotropic_t = LambertianParams<LS, Iso, scalar_type>;
+    using params_anisotropic_t = LambertianParams<LS, Aniso, scalar_type>;
+
+
+    static this_t create()
+    {
+        this_t retval;
+        // nothing here, just keeping in convention with others
+        return retval;
+    }
+
+    static this_t create(NBL_CONST_REF_ARG(SBxDFCreationParams<scalar_type, spectral_type>) params)
+    {
+        return create();
+    }
+
+    void init(NBL_CONST_REF_ARG(SBxDFCreationParams<scalar_type, spectral_type>) params)
+    {
+        // do nothing
+    }
+
+    scalar_type __eval_pi_factored_out(scalar_type maxNdotL)
+    {
+        return maxNdotL;
+    }
+
+    scalar_type eval(NBL_CONST_REF_ARG(params_isotropic_t) params)
+    {
+        return __eval_pi_factored_out(params.getNdotL()) * numbers::inv_pi<scalar_type>;
+    }
+    scalar_type eval(NBL_CONST_REF_ARG(params_anisotropic_t) params)
+    {
+        return __eval_pi_factored_out(params.getNdotL()) * numbers::inv_pi<scalar_type>;
+    }
+
+    sample_type generate_wo_clamps(NBL_CONST_REF_ARG(anisotropic_interaction_type) interaction, NBL_CONST_REF_ARG(vector<scalar_type, 2>) u)
+    {
+        ray_dir_info_type L;
+        L.direction = sampling::ProjectedHemisphere<scalar_type>::generate(u);
+        return sample_type::createFromTangentSpace(interaction.getTangentSpaceV(), L, interaction.getFromTangentSpace());
```

Comment:
what do we need `V` for in an output sample ?

(There's another comment somewhere where I ask you about what is `VdotL` doing in the `SLightSample` and the concept)